src/site/_includes/components/tags/Image.js

```diff
@@ -0,0 +1,211 @@
+ * {%
+ *  Figure
+ *  src="./label-a11y-lint.png",
+ *  alt="hello",
+ *  type="fullbleed screenshot",
+ *  maxWidth="300px"
+ * %}
+ *  Figcaption text goes here! Can be `markdown`.
+ * {% endFigure %}
+ *
+ */
+
+const path = require('path');
+const {oneLine} = require('common-tags');
+const {ifDefined} = require('../helpers');
+const md = require('markdown-it')();
+const stripLanguage = require('../../../_filters/strip-language');
+const imageCdn = require('../../../_data/site').imageCdn;
+
+/* eslint-disable no-invalid-this, max-len */
+
+// -----------------------------------------------------------------------------
+// IMAGE
+// -----------------------------------------------------------------------------
+
+/**
+ * Takes a path to an image and converts it to an image CDN path if we're in
+ * a production environment.
+ * @param {!string} src A path to an image asset.
+ * @param {!Object} ctx An eleventy context object for the current page.
+ * @return {string} An image path. May be converted to an image CDN path if
+ * it's a production environment.
+ */
+function getImagePath(src, ctx) {
+  let imagePath = src;
+  if (process.env.ELEVENTY_ENV === 'prod') {
+    imagePath = path.join(stripLanguage(ctx.page.url), src);
+    imagePath = new URL(imagePath, imageCdn).href;
+  }
+  return imagePath;
+}
+
+/**
+ * Render an image element as an HTML string.
+ * @param {!Object} args An args object containing src and optional alt and
+ * maxWidth.
+ * @param {!Object} ctx An eleventy context object.
+ * @return {string}
+ */
+function renderImage({src, alt, maxWidth}, ctx) {
+  const imagePath = getImagePath(src, ctx);
+  let style;
+  if (maxWidth) {
+    style = `max-width: ${maxWidth};`;
+  }
+  // Using oneLine seems to make Nunjucks' SafeString function happy.
+  return oneLine`
+    <img
+      src="${imagePath}"
+      alt="${alt}"
+      ${ifDefined('style', style)}
+    />
+  `;
+}
+
+/**
+ * Define an Image shortcode to be used in page templates.
+ * @param {!Object} nunjucksEngine
+ * @return {function}
+ */
+const Image = (nunjucksEngine) => {
+  return new function() {
+    this.tags = ['Image'];
+
+    this.parse = function(parser, nodes, lexer) {
+      const tok = parser.nextToken();
+
+      const args = parser.parseSignature(null, true);
+      parser.advanceAfterBlockEnd(tok.value);
+
+      return new nodes.CallExtensionAsync(this, 'run', args);
+    };
+
+    this.run = function({ctx}, args, callback) {
+      const ret = new nunjucksEngine.runtime.SafeString(
+        renderImage(args, ctx)
+      );
+      callback(null, ret);
+    };
+  }();
+};
+
+// -----------------------------------------------------------------------------
+// FIGURE
+// -----------------------------------------------------------------------------
+
+/**
+ * Render a figcaption element as an HTML string.
+ * @param {?string} caption A markdown string of caption text.
+ * @param {Array<string>} figCaptionClasses An Array of classes to apply to the
```

Comment:
`!Array<string>` is non-nullable, which is what you want: `null` would be invalid here since you call a method on the array